src/test/java/com/notably/logic/parser/suggestion/DeleteSuggestionCommandParserTest.java

```diff
@@ -91,374 +46,310 @@ public static void setUp() {
-                null);
-
-        List<SuggestionItem> expectedSuggestions = new ArrayList<>();
-        expectedSuggestions.add(cs2103);
-        expectedSuggestions.add(cs2103Week1);
-        expectedSuggestions.add(cs2103Week2);
-        expectedSuggestions.add(cs2103Week3);
-        expectedSuggestions.add(cs2103Week1Lecture);
+        // Expected suggestions
+        List<SuggestionItem> expectedSuggestions = SuggestionTestUtil.getExpectedSugForCs2103tPathInput();
 
+        // Actual suggestions
         List<SuggestionItem> suggestions = model.getSuggestions();
 
-        // check display text
-        for (int i = 0; i < expectedSuggestions.size(); i++) {
-            SuggestionItem suggestion = suggestions.get(i);
-            SuggestionItem expectedSuggestion = expectedSuggestions.get(i);
-            assertEquals(expectedSuggestion.getProperty("displayText"), suggestion.getProperty("displayText"));
-        }
-
-        List<String> expectedInputs = new ArrayList<>();
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103Week1.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103Week2.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103Week3.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + PREFIX_TITLE + " " + toCs2103Week1Lecture.getStringRepresentation());
-
-        for (int i = 0; i < expectedInputs.size(); i++) {
-            SuggestionItem suggestionItem = suggestions.get(i);
-            String expectedInput = expectedInputs.get(i);
-            suggestionItem.getAction().run();
-            String input = model.getInput();
-            assertEquals(expectedInput, input);
-        }
+        // Test suggestions
+        SuggestionTestUtil.testSuggestions(expectedSuggestions, suggestions);
+
+        // Expected inputs
+        List<String> expectedInputs = SuggestionTestUtil.getExpectedInputsForCs2103tPathInput(userInputWithoutPath);
+
+        // Test inputs
+        SuggestionTestUtil.testInputs(expectedInputs, suggestions, model);
     }
 
     @Test
-    public void parse_correctedAbsolutePathWithoutPrefix_returnsDeleteSuggestionCommand() {
-        String arg = "/CS2104";
-        model.setInput(COMMAND_WORD + " " + arg);
-        Optional<? extends SuggestionCommand> commandCorrectPath = deleteSuggestionCommandParser.parse(arg);
-        assertTrue(commandCorrectPath.get() instanceof DeleteSuggestionCommand);
+    public void parse_correctCmdcorrectRelativePathWithPrefix_returnsDeleteSuggestionCommand() {
+        String userInputWithoutPath = COMMAND_WORD + " " + PREFIX_TITLE + " ";
+        String userInput = userInputWithoutPath + RELATIVE_PATH_TO_CS2103T;
+        String arg = userInput.replace(COMMAND_WORD, "");
```

Comment:
I'd say we do away with this kind of `replace` calls, since it makes it more confusing. You can just hard code the `arg` string, I believe?